app/src/main/java/de/westnordost/streetcomplete/quests/address/AddAddressStreetForm.kt

```diff
@@ -0,0 +1,179 @@
+package de.westnordost.streetcomplete.quests.address
+
+import android.view.Menu
+import android.view.View
+import android.widget.EditText
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.widget.PopupMenu
+import de.westnordost.osmapi.map.data.LatLon
+import de.westnordost.streetcomplete.Injector
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.meta.AbbreviationsByLocale
+import de.westnordost.streetcomplete.data.osm.elementgeometry.ElementGeometry
+import de.westnordost.streetcomplete.data.osm.elementgeometry.ElementPointGeometry
+import de.westnordost.streetcomplete.data.osm.elementgeometry.ElementPolygonsGeometry
+import de.westnordost.streetcomplete.data.osm.elementgeometry.ElementPolylinesGeometry
+import de.westnordost.streetcomplete.quests.LastPickedValuesStore
+import de.westnordost.streetcomplete.quests.localized_name.AAddLocalizedNameForm
+import de.westnordost.streetcomplete.quests.OtherAnswer
+import de.westnordost.streetcomplete.quests.localized_name.AddLocalizedNameAdapter
+import de.westnordost.streetcomplete.quests.localized_name.LocalizedName
+import de.westnordost.streetcomplete.quests.localized_name.data.RoadNameSuggestionsDao
+import de.westnordost.streetcomplete.util.TextChangedWatcher
+import java.util.*
+import javax.inject.Inject
+
+class AddAddressStreetForm : AAddLocalizedNameForm<AddressStreetAnswer>() {
+    private var textField: EditText? = null
+    private var isPlaceName = false
+    private var defaultName = ""
+
+    override fun onClickOk(names: List<LocalizedName>) {
+        val possibleAbbreviations = LinkedList<String>()
+        for ((languageCode, name) in adapter.localizedNames) {
+            val locale = if(languageCode.isEmpty()) countryInfo.locale else Locale(languageCode)
+            val abbr = abbreviationsByLocale.get(locale)
+            val containsAbbreviations = abbr?.containsAbbreviations(name) == true
+
+            if (name.contains(".") || containsAbbreviations) {
+                possibleAbbreviations.add(name)
+            }
+        }
+
+        // TODO, not sure if I should be saving all the names (is there every going to be more than one?)
```

Comment:
Saving the "main" name (`addr:street`) is enough, because the localizations if any can be added back via the road name suggestions database.

Reply:
ok, I removed the comment suggesting that it might need to do so in https://github.com/westnordost/StreetComplete/pull/1782/commits/7bbf0c31c7d0954980226831282b2dd40980eef8

Reply:
Removed TODO comment in https://github.com/westnordost/StreetComplete/pull/1782/commits/7bbf0c31c7d0954980226831282b2dd40980eef8